src/hierarchies/base_hierarchy.h

```diff
@@ -44,6 +44,24 @@ class BaseHierarchy : public AbstractHierarchy {
     return out;
   }
 
+  //! Returns an independent, data-less copy of this object
+  virtual std::shared_ptr<AbstractHierarchy> deep_clone() const override {
+    auto out = std::make_shared<Derived>(static_cast<Derived const &>(*this));
+    out->clear_data();
+    out->clear_summary_statistics();
+    out->reset_hypers();
+    out->reset_prior();
+    std::shared_ptr<google::protobuf::Message> new_prior(prior->New());
+    new_prior->CopyFrom(*prior.get());
+    out->get_mutable_prior()->CopyFrom(*new_prior.get());
+    out->initialize();
+    return out;
+  }
```

Comment:
Why do you say empty copy? We deep-copy the prior instead of having it point to the same object. I think that we can do the same for the hypers to avoid confusion.

Reply:
My b, I misunderstood what `New()` does. We should definitely copy both objects in the same way if possible